packages/core/src/_common/base/api_middleware.js

```diff
@@ -1,9 +1,94 @@
+const datadogLogs = require('@datadog/browser-logs').datadogLogs;
+const formatDate = require('@deriv/shared').formatDate;
+const formatTime = require('@deriv/shared').formatTime;
+
+const datadog_client_token_logs = process.env.DATADOG_CLIENT_TOKEN_LOGS ?? '';
+const is_production = process.env.CIRCLE_JOB === 'release_production';
+const is_staging = process.env.CIRCLE_JOB === 'release_staging';
+const datadog_session_sample_rate = +process.env.DATADOG_SESSION_SAMPLE_RATE_LOGS ?? 1;
```

Comment:
it is a draft PR, but don't forget to add `DATADOG_CLIENT_TOKEN_LOGS `and `DATADOG_SESSION_SAMPLE_RATE_LOGS`  in DefinePlugin inside plugins in webpack. currently it is added only in bot-web-ui package as this is used only there.